### PR TITLE
feat: support mTLS connection to ETCD

### LINKED ISSUE #681: feature: enhance ETCD config
relate to  #620

  - [x] Support https connection to ETCD

  - [ ] Support mTLS connection to ETCD

  - [x] Support authentication for ETCD

### PR DESCRIPTION
**Pull request by nic-chen**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

- Related issues
close #681

___
### New feature or improvement
- Describe the details and related test reports.

support mTLS connection to ETCD

### CODE REVIEW

File: api/test/shell/cli_test.sh [hunk 1]
```diff
@@ -338,3 +338,40 @@ if [[ `echo ${resp} | grep -c "${GITHASH}"` -ne '1' ]]; then
 fi
 
 check_logfile
+
+
+# mtls test
+
+wget https://github.com/etcd-io/etcd/releases/download/v3.4.14/etcd-v3.4.14-linux-amd64.tar.gz
+
+tar zxvf etcd-v3.4.14-linux-amd64.tar.gz && cd etcd-v3.4.14-linux-amd64
```
Comment (starsz) [inline api/test/shell/cli_test.sh:347]: Can this be done at github action?

File: api/test/shell/cli_test.sh [hunk 2]
```diff
@@ -338,3 +338,40 @@ if [[ `echo ${resp} | grep -c "${GITHASH}"` -ne '1' ]]; then
...
+
+./etcd --name infra0 --data-dir infra0 \
+  --client-cert-auth --trusted-ca-file=$(pwd)/test/certs/mtls_ca.crt --cert-file=$(pwd)/test/certs/mtls_server.crt --key-file=$(pwd)/test/certs/mtls_server.key \
+  --advertise-client-urls https://127.0.0.1:3379 --listen-client-urls https://127.0.0.1:3379 --listen-peer-urls http://0.0.0.0:3380 &
+
```
Comment (starsz) [inline api/test/shell/cli_test.sh:352]: We should stop the etcd after test.

File: api/test/shell/cli_test.sh [hunk 3]
```diff
@@ -338,3 +338,40 @@ if [[ `echo ${resp} | grep -c "${GITHASH}"` -ne '1' ]]; then
...
+fi
+
+./manager-api &
+sleep 3
+
+# validate process is right by requesting login api
+resp=$(curl http://127.0.0.1:9000/apisix/admin/user/login -H "Content-Type: application/json" -d '{"username":"admin", "password": "admin"}')
+token=$(echo "${resp}" | sed 's/{/\n/g' | sed 's/,/\n/g' | grep "token" | sed 's/:/\n/g' | sed '1d' | sed 's/}//g'  | sed 's/"//g')
+if [ -z "${token}" ]; then
+    echo "login failed"
```
Comment (starsz) [inline api/test/shell/cli_test.sh:375]: `login failed` is already existed.
We should add extra info to distinguish them.

**Review by starsz — CHANGES_REQUESTED** [review]

File: api/test/shell/cli_test.sh (hunk 1 above)
Comment (nic-chen) [reply api/test/shell/cli_test.sh]: fixed.

File: api/test/shell/cli_test.sh (hunk 2 above)
Comment (nic-chen) [reply api/test/shell/cli_test.sh:349]: fixed.

File: api/test/shell/cli_test.sh (hunk 3 above)
Comment (nic-chen) [reply api/test/shell/cli_test.sh]: updated.

**Review by gxthrj — APPROVED** [review]

**Pull request merged**